src/main/java/ru/skypro/hw/hwCollections3/controller/DepartmentController.java

```diff
@@ -0,0 +1,47 @@
+package ru.skypro.hw.hwCollections3.controller;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import ru.skypro.hw.hwCollections3.domein.Employee;
+import ru.skypro.hw.hwCollections3.service.DepartmentService;
+import ru.skypro.hw.hwCollections3.service.DepartmentServiceImpl;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping(path = "/department")
+public class DepartmentController {
+
+    private final DepartmentService departmentService;
+    public DepartmentController(DepartmentService departmentService){
+        this.departmentService = departmentService;
+    }
+
+
+    @GetMapping(path = "/min-salary")
+    public Employee getMinSalary(@RequestParam("departmentId") Integer dept){
+        return departmentService.getMinSalary(dept);
+    }
+
+    @GetMapping(path = "/max-salary")
+    public Employee getMaxSalary(@RequestParam("departmentId") Integer dept){
+        return departmentService.getMaxSalary(dept);
+    }
+
+    @GetMapping(path = "/all-dept")
```

Comment:
путь должен быть /all, как написано в задании. Чтобы разрешить конфликт для спринга надо указать параметр в params
```suggestion
    @GetMapping(path = "/all", params = "departmentId")
```

Reply:
Не попадалось такое решение. Спасибо